File: C-Plus-Plus/ValueOfPi.cpp
#include <bits/stdc++.h> 
#define INTERVAL 10000 
using namespace std; 

int main() 
{ 
    int interval, i; 
    double rand_x, rand_y, origin_dist, pi; 
    int circle_points = 0, square_points = 0; 

    srand(time(NULL)); 

    // Total Random numbers generated = possible x 
    // values * possible y values 
    for (i = 0; i < (INTERVAL * INTERVAL); i++) { 

        // Randomly generated x and y values 
        rand_x = double(rand() % (INTERVAL + 1)) / INTERVAL; 
        rand_y = double(rand() % (INTERVAL + 1)) / INTERVAL; 

        // Distance between (x, y) from the origin 
        origin_dist = rand_x * rand_x + rand_y * rand_y; 

        // Checking if (x, y) lies inside the define 
        // circle with R=1 
        if (origin_dist <= 1) 
            circle_points++; 

        // Total number of points generated 
        square_points++; 

        // estimated pi after this iteration 
        pi = double(4 * circle_points) / square_points; 
    } 

    // Final Estimated Value 
    cout << "\nFinal Estimation of Pi = " << pi; 

    return 0; 
} 
/*
Final Estimation of Pi = 3.14167
time complexity: O(INTERVAL*2^INTERVAL)
space complexity: O(1)
*/

File: Python/valueofPi.py
import random
PERIOD = 1000
circle_coords = 0
square_coords = 0

# Count of Random number generated will be in [0,xVal * yVal]
for i in range(PERIOD**2):
   rand_x = random.uniform(-1,1)
   rand_y = random.uniform(-1,1)
   dist = rand_x**2 + rand_y**2
   # Check if x & y inside the circle
   if dist <= 1:
      circle_coords += 1
   square_coords += 1
   pi = 4* circle_coords/square_coords

print('Estimated Value of Pi using Monte Carlo is',pi)
"""
Final Estimation of Pi - 3.140484
Time Complexity - O(PERIOD* 2^PERIOD)
Space Complexity - O(1)
"""